### PR TITLE
Prefer split patterns from directories over split patterns from filenames

### PR DESCRIPTION
**Pull request by polinaeterna**
related to https://github.com/huggingface/datasets/issues/4895

todo:

- [ ] test

### CODE REVIEW

File: src/datasets/data_files.py [hunk 1]
```diff
@@ -73,10 +73,10 @@ class Url(str):
     Split.TRAIN: ["**"],
 }
 
-ALL_SPLIT_PATTERNS = [SPLIT_PATTERN_SHARDED]
+ALL_SPLIT_PATTERNS = [SPLIT_PATTERN_SHARDED]  # TODO: should this be changed?
```
Comment (lhoestq) [inline src/datasets/data_files.py:76]: I think it's fine this way. This corresponds to the patterns that contain custom split names (other than train/dev/test)

**Review by lhoestq — COMMENTED** [review]
Thanks ! adding a test just for this is maybe overkill but as you want

**Comment by polinaeterna** [comment]
@mariosasko sure! could you please approve it?

**Comment by shaneacton** [comment]
Hi there! I have installed 5.2.3.dev0, which should have this fix. Unfortunately, I am still getting the error:
`ValueError: Unknown split "validation". Should be one of ['train'].` When I call `load_dataset("csv", data_files=files, split=split)`

Any help would be greatly appreciated!

**Pull request merged**